app/assets/javascripts/Patient.js

```diff
@@ -0,0 +1,162 @@
+}
+
+PatientSchema.methods.family_history = function family_history(params, callback) {
+  return this.get_history_elements({'category': 'family_history'});
+}
+
+PatientSchema.methods.functional_statuses = function functional_statuses(params, callback) {
+  return this.get_history_elements({'category': 'functional_status'});
+}
+
+PatientSchema.methods.immunizations = function immunizations(params, callback) {
+  return this.get_history_elements({'category': 'immunization'});
+}
+
+PatientSchema.methods.interventions = function interventions(params, callback) {
+  return this.get_history_elements({'category': 'intervention'});
+}
+
+PatientSchema.methods.laboratory_tests = function laboratory_tests(params, callback) {
+  return this.get_history_elements({'category': 'laboratory_test'});
+}
+
+PatientSchema.methods.medical_equipment = function medical_equipment(params, callback) {
+  return this.get_history_elements({'category': 'medical_equipment'});
+}
+
+PatientSchema.methods.medications = function medications(params, callback) {
+  return this.get_history_elements({'category': 'medication'});
+}
+
+PatientSchema.methods.physical_exams = function physical_exams(params, callback) {
+  return this.get_history_elements({'category': 'physical_exam'});
+}
+
+PatientSchema.methods.preferences = function preferences(params, callback) {
+  return this.get_history_elements({'category': 'preference'});
+}
+
+PatientSchema.methods.provider_characteristics = function provider_characteristics(params, callback) {
+  return this.get_history_elements({'category': 'provider_characteristic'});
+}
+
+PatientSchema.methods.procedures = function procedures(params, callback) {
+  return this.get_history_elements({'category': 'procedure'});
+}
+
+PatientSchema.methods.results = function results(params, callback) {
+  return this.get_history_elements({'category': 'result'});
+}
+
+PatientSchema.methods.risk_category_assessments = function risk_category_assessments(params, callback) {
+  return this.get_history_elements({'category': 'risk_category_assessment'});
+}
+
+PatientSchema.methods.social_history = function social_history(params, callback) {
+  return this.get_history_elements({'category': 'social_history'});
+}
+
+PatientSchema.methods.substances = function substances(params, callback) {
+  return this.get_history_elements({'category': 'substance'});
+}
+
+PatientSchema.methods.symptoms = function symptoms(params, callback) {
+  return this.get_history_elements({'category': 'symptom'});
+}
+
+PatientSchema.methods.system_characteristics = function system_characteristics(params, callback) {
+  return this.get_history_elements({'category': 'system_characteristic'});
+}
+
+PatientSchema.methods.transfers = function transfers(params, callback) {
+  return this.get_history_elements({'category': 'transfer'});
+}
+
+PatientSchema.methods.vital_signs = function vital_signs(params, callback) {
+  return this.get_history_elements({'category': 'vital_sign'});
+}
+
```

Comment:
If the plan is to use this Patient Model with the CQL execution_engine, then we will need to add a "find_records" function, similar to what we have in the cql_qdm_patientapi, in the "cqlpatient.js.coffee" file.

Reply:
Right, same with `codes` on datatypes.